common/persistence/client/factory.go

```diff
@@ -119,8 +122,8 @@ func (f *factoryImpl) NewTaskManager() (p.TaskManager, error) {
 	}
 
 	result := p.NewTaskManager(taskStore, f.serializer)
-	if f.ratelimiter != nil {
-		result = p.NewTaskPersistenceRateLimitedClient(result, f.ratelimiter, f.logger)
+	if f.systemRateLimiter != nil && f.namespaceRateLimiter != nil {
```

Comment:
nit: requiring both seems might be surprising to the caller. But no strong opinion.

Reply:
Right. I will combine these two rate limiters into a single ScopedRatelimiter object. That should fix this.